inst/@sym/solve.m

```diff
@@ -129,22 +149,23 @@
             '        symbols.append(arg)'
             '#'
             'if len(symbols) > 0:'
-            '    (vars, solns) = sp.solve(eqs, symbols, set=True)'
+            '    d = sp.solve(eqs, *symbols, set=True)'
             'else:'
-            '    (vars, solns) = sp.solve(eqs, set=True)'
+            '    d = sp.solve(eqs, set=True)'
             '#'
-            'd = []'
+            'if not isinstance(d, (list, tuple)):'
+            '    return d,'
+            '(vars, solns) = d'
+            'q = []'
             'for (i, var) in enumerate(vars):'
-            '    d.append(sp.Matrix([t[i] for t in solns]))'
-            'return d,' };
+            '    q.append(sp.Matrix([t[i] for t in solns]))'
+            'return q,' };
 
     out = python_cmd (cmd, varargin{:});
 
-    varargout = out;
+    assert( isequal( nargout, length(out)), ['This system have ' num2str(length(out)) ' solution, favor set correct numbers of output.']);
 
-    if (length(out) ~= nargout)
-      warning('solve: number of outputs did not match solution vars');
-    end
```

Comment:
This can be removed, if the length of nargout != length of out octave will handle the error, so before that we can send a personalized error:
```
>> [X Y Z]=solve(x^2 == 4, x + y == 1, x, y);
warning: solve: number of outputs did not match solution vars
warning: called from
    solve at line 146 column 7
error: element number 3 undefined in return list
```